### ISSUE TITLE
ES2015 / ES.next benchmarks for Node.js?

### ISSUE THREAD

**Issue by bmeurer** [nodejs/benchmarking#60]
The V8 team is looking to close the gap on ES.next vs ES5 performance, i.e. make ES2015 and beyond features as fast as their (naive) counterparts in ES5. There's a [preliminary performance plan](https://docs.google.com/document/d/1EA9EbfnydAmmU_lM8R_uEMQ-U_v4l9zulePSBkeYWmY) available. This work will be beneficial for the Node community in particular, as users have a dedicated environment, i.e. don't need to transpile to make it work on older browsers.

Is there any interest to add particular ES.next benchmarks? Maybe with special focus on features that matter for Node? We have a partial ES6 port of Octane, but it's unclear how useful that is for real world Node ES6 performance.

**Comment by fhinkel** [comment]
cc @nodejs/ctc

**Comment by yosuke-furukawa** [comment]
I have an interest in the full ES2015 node is so slower than current one. I will create the node repository and send performance result on this thread first.

**Comment by michaelbpaulson** [comment]
@nodejs/benchmarking it'll be best if we could generate a sophisticated enough test that the ES2015 features are exhaustively tested.  I can start brain storming some ideas on what to build and we can address this in the next meeting

**Comment by Fishrock123** [comment]
related: https://github.com/nodejs/node/issues/8637

**Comment by mhdawson** [comment]
Agreed, an ES.next benchmark would be interesting and it would be better if it was turned to reflect Node.js usage.  I we can end up with something that has a small number of numbers that we can track/report that would be good.  @michaelbpaulson sounds good.  I'll look at setting up a time for the next meeting.

**Comment by mhdawson** [comment]
@bmeurer is this still relevant ?

**Comment by bmeurer** [comment]
Definitely. In the meantime Apple created an ES6 benchmark called [ARES-6](http://browserbench.org/ARES-6/), but it is not very representative of real workloads (except for the Babylon case) currently. And it doesn't contain any post ES6 features like `async`/`await`.
